### ISSUE TITLE
WhatsApp Update Available

### ISSUE THREAD

**Issue by canuckbrian** [ramboxapp/community-edition#2217]
WhatsApp is always saying there's an update available. I click to update, the service reloads, and it's still saying there's an update.

1) Open Rambox
2) Go to existing WhatsApp service tab or create a new one and sign in
3) Click on "Update available" bar above chats list
4) Service reloads
5) Update notification is still there

In the previous Rambox version 0.6.4, it would display this notice each time Rambox was opened, and after clicking to update the notice would go away, but now it's there all the time.
	-
	> Rambox 0.6.6
	> Electron 4.1.4
	> win32 x64 10.0.17763

**Comment by Corchoneitor** [comment]
Hi @canuckbrian!

Also happens every time you disable-enable the service, and when clearing cache.

We'll have a look at it.

**Comment by marcosrocha85** [comment]
I know it's not bug related itself, but my Whatsapp tab isn't playing any sound either besides that bug.

**Comment by maxerbox** [comment]
It's maybe related again to the user agent.

**Comment by maxerbox** [comment]
> I know it's not bug related itself, but my Whatsapp tab isn't playing any sound either besides that bug.
Check out if the do not disturb mode is enabled, and if the tab is muted

**Comment by canuckbrian** [comment]
So funny enough, it's no longer doing this. I started Rambox last night, and for shits n' giggles I clicked on the Update notification and low and behold it went away after the service reloaded. I've restarted Rambox and the Whatsapp service many times since and it hasn't come back. Seems to be working as expected now.

**Comment by marcosrocha85** [comment]
> 
> 
> > I know it's not bug related itself, but my Whatsapp tab isn't playing any sound either besides that bug.
> 
> Check out if the do not disturb mode is enabled, and if the tab is muted

Sorry mate, but disturb mode is disabled and the tab isn't muted as well.

**Comment by marcosrocha85** [comment]
> 
> 
> > I know it's not bug related itself, but my Whatsapp tab isn't playing any sound either besides that bug.
> 
> Check out if the do not disturb mode is enabled, and if the tab is muted

Thanks mate, disturb mode was enabled and I ain't used it before.

**Comment by Corchoneitor** [comment]
> So funny enough, it's no longer doing this. I started Rambox last night, and for shits n' giggles I clicked on the Update notification and low and behold it went away after the service reloaded. I've restarted Rambox and the Whatsapp service many times since and it hasn't come back. Seems to be working as expected now.

Try to disable-enable the service, it's still happening.

**Comment by Jendker** [comment]
Same here, I can also observe it.

**Comment by marcosrocha85** [comment]
Thank you. Update v0.6.7 solved the problem.

**Comment by canuckbrian** [comment]
> Thank you. Update v0.6.7 solved the problem.

Same here. 0.6.7 resolved this issue for me too. Closing issue.

**Comment by Jendker** [comment]
Is this actually fixed?

I still get this "Update available" bar in WhatsApp. If I click it, it goes away until the next Rambox restart. Release note of update 0.6.7 also didn't mention any fix to this problem.

**Comment by DerAlbertCom** [comment]
same here was still present in 0.6.7. Now I removed WhatsApp, closed RamBox, added WhatsApp. Problem seems gone (i did a couple of restarts von Rambox,, no update message). Will update this after the next Update Message.

**Comment by DerAlbertCom** [comment]
Sadly, this does not help. Now getting Updates Messages over and over again.

**Comment by MosEisley34** [comment]
I am too getting these messages now.

**Comment by pradyunsg** [comment]
I'll note that I saw Station has similar issues with Whatsapp as well. Might be interesting to try to see what they did to fix this.

**Comment by sp00n** [comment]
I have the same thing happening in 0.67. Every time I start Rambox the green message that a WhatsApp update is available appears. Sometimes it goes away when I click the message. Sometimes it goes away when I click the message multiple times. Sometimes (like as right now) it doesn't go away no matter how often I click the message.

**Comment by adarmalik** [comment]
Same problem here. Installed 0.6.7 from snap on neon (Ubuntu 18.04). Sometimes it disappears if I disable the service, restart and re-enable it. But it keeps coming back.

**Comment by DanielAlcon** [comment]
same problem here, rambox 0.6.9 in mac os mojave 10.14.5

**Comment by izolini** [comment]
0.6.9 has same problem on Win10.

**Comment by sp00n** [comment]
0.6.9 is actually a lot better than previous versions for me. At least I can get rid of the update message by clicking on it.

**Comment by jerry1970** [comment]
The problem is still there in 0.6.9. When I open web.whatsapp.com in a browser window I don't get the message and I get a message (same space where the update message is but now with a blue background) telling me about desktop notifications.

It is like Rambox is requesting an older version of the whatsapp UI. Maybe a version is stored in a cookie or localstorage somewhere and within Rambox that does not get updated?

**Comment by stefanuddenberg** [comment]
WhatsApp Web no longer works; now it asks you to update the web browser (Google Chrome).

**Comment by eliaszervudakis** [comment]
The "Update Available" issue still exists (0.6.9.).

The version number of the WhatsApp web instance (Settings->Help) does not change after pressing the "Update available" button.

**Comment by wolfposd** [comment]
still remains in 0.7.0

**Comment by dekaru** [comment]
Still remained in 0.7.2

**Comment by YerayAlonso** [comment]
Add the following code in the "advanced functions" box at the whatsapp settings to hide that alert:

```javascript
(()=>{ let css=`
._3O0po {
    display: none;
}
`;

var ss = document.createElement("style");
ss.type = "text/css"; ss.innerHTML = css;
document.getElementsByTagName("head")[0].appendChild(ss); })();
```

**Comment by Tsakatac** [comment]
Hello,

I have the same issue also.

> Rambox 0.7.9
> Electron 11.4.10
> win32 x64 10.0.19043

**Comment by nickchomey** [comment]
It's worth mentioning that Hamsket doesn't have this problem, so perhaps there is something in their code that can be useful here? (I'm using rambox because Hamsket has a bunch of far more annoying problems).

**Comment by localhorst** [comment]
> Add the following code in the "advanced functions" box at the whatsapp settings to hide that alert:
> 
> ```js
> (()=>{ let css=`
> ._3O0po {
>     display: none;
> }
> `;
> 
> var ss = document.createElement("style");
> ss.type = "text/css"; ss.innerHTML = css;
> document.getElementsByTagName("head")[0].appendChild(ss); })();
> ```

Thanks, but this doesn't work for me no more.

**Comment by francwalter** [comment]
> Thanks, but this doesn't work for me no more.

There should be some way to find the right class name, as that "_3O0po"  is maybe dynamically created and subject to change often. It doesnt work for me neither.

**Comment by francwalter** [comment]
> It's worth mentioning that Hamsket doesn't have this problem, so perhaps there is something in their code that can be useful here? (I'm using rambox because Hamsket has a bunch of far more annoying problems).
This is wrong. I use Hamsket and I have it often.

**Comment by nickchomey** [comment]
@francwalter yes, it has come back in Hamsket. But when I wrote that it wasn't happening. I noticed it come and go a while ago as well. I assume it has something to do with whichever electron or chrome versions are being used.

Anyway, yes, Hamsket seems more active so I've switched back to it now as well.

**Comment by francwalter** [comment]
> Anyway, yes, Hamsket seems more active so I've switched back to it now as well. 

Do _you_ know how to get grip of that content of Whatsapp service with the developer tools?
I see only the webview tag and the url from Whatsapp web but not the content so I could find the actual class of the update box :(

**Comment by nickchomey** [comment]
I have no idea how to fix any of this. I just use Hamsket or rambox based on which has the fewest bugs at any given time. Hamsket seems to be ahead for now.

**Comment by chiu0602** [comment]
Thanks @YerayAlonso's inspiration!
```
(()=>{ let css=`
header + span {
    display: none !important;
}
`;

var ss = document.createElement("style");
ss.type = "text/css"; ss.innerHTML = css;
document.getElementsByTagName("head")[0].appendChild(ss); })();
```
